### PR TITLE
Convenience methods to serialize attributes to JSON string and other cleanup

### PR DESCRIPTION
**Pull request by aarsilv**
_Eppo Internal:_ 
🎟️ **Ticket 1:** [FF-1648 - Add more isXXX checks for EppoValue (and ideally a JSON serialize)](https://linear.app/eppo/issue/FF-1648/add-more-isxxx-checks-for-eppovalue-and-ideally-a-json-serialize)
🎟️ **Ticket 2:** [FF-1764 - Log "Bandit Key" in the bandit logger instead of "Variation"](https://linear.app/eppo/issue/FF-1764/log-bandit-key-in-the-bandit-logger-instead-of-variation)

For a user implementing our loggers, they will most likely need to serialize attributes to a JSON string when storing subject attributes for the assignment and subject and action numeric and categorical attributes for the bandit assignment.

This PR adds two new static utility functions to `EppoAttributes`:
* `serializeAttributesToJSONString()`
* `serializeNonNullAttributesToJSONString()`
There is also a convenience instance method added to `EppoAttributes`:
* `serializeAttributesToJSONString()`

For better logger clarity, we've also renamed `variation` to `banditKey`.

Example usage by the Bandit Proxy:
```java
private static EppoClient initClient() {
    String apiKey = dotenv.get("EPPO_SDK_KEY");
    EppoClientConfig config = EppoClientConfig.builder()
      .apiKey(apiKey)
      .assignmentLogger(assignmentLogData -> {
        String sql = "INSERT INTO bandit_test_assignments " +
          "(timestamp, flag_name, experiment_name, variation_value, subject_key, subject_attributes) " +
          "SELECT ?, ?, ?, ?, ?, parse_json(?)";

        try (PreparedStatement statement = snowflakeConnection.prepareStatement(sql)) {
          statement.setTimestamp(1, new Timestamp(assignmentLogData.timestamp.getTime()));
          statement.setString(2, assignmentLogData.featureFlag);
          statement.setNull(3, Types.NULL); // Experiment hard coded to null for now
          statement.setString(4, assignmentLogData.variation);
          statement.setString(5, assignmentLogData.subject);
          if (assignmentLogData.subjectAttributes == null) {
            statement.setNull(6, Types.NULL);
          } else {
             // 👇 USED HERE
            statement.setString(6, EppoAttributes.serializeNonNullAttributesToJSONString(assignmentLogData.subjectAttributes));
          }

          statement.executeUpdate();
          System.out.println("Logged assignment to bandit_test_assignments");
        } catch (SQLException e) {
          throw new RuntimeException("Unable to log assignment "+e.getMessage(), e);
        }
      })
      .banditLogger(banditLogData -> {
        String sql = "INSERT INTO bandit_test_bandit_assignments " +
          "(timestamp, experiment_name, variation_value, subject_key," +
          " action_name, action_probability, model_version," +
          " subject_numeric_attributes, subject_categorical_attributes," +
          " action_numeric_attributes, action_categorical_attributes) " +
          "SELECT ?, ?, ?, ?," +
          " ?, ?, ?," +
          " parse_json(?), parse_json(?)," +
          " parse_json(?), parse_json(?)";

        try (PreparedStatement statement = snowflakeConnection.prepareStatement(sql)) {
          statement.setTimestamp(1, new Timestamp(banditLogData.timestamp.getTime()));
          statement.setString(2, banditLogData.experiment);
          statement.setString(3, banditLogData.banditKey);
          statement.setString(4, banditLogData.subject);
          statement.setString(5, banditLogData.action);
          statement.setDouble(6, banditLogData.actionProbability);
          statement.setString(7, banditLogData.modelVersion);
          if (banditLogData.subjectNumericAttributes == null) {
            statement.setNull(8, Types.NULL);
          } else {
            // 👇 USED HERE
            statement.setString(8, EppoAttributes.serializeNonNullAttributesToJSONString(banditLogData.subjectNumericAttributes));
          }
          if (banditLogData.subjectCategoricalAttributes == null) {
            statement.setNull(9, Types.NULL);
          } else {
            // 👇 USED HERE
            statement.setString(9, EppoAttributes.serializeNonNullAttributesToJSONString(banditLogData.subjectCategoricalAttributes));
          }
          if (banditLogData.actionNumericAttributes == null) {
            statement.setNull(10, Types.NULL);
          } else {
            // 👇 USED HERE
            statement.setString(10, EppoAttributes.serializeNonNullAttributesToJSONString(banditLogData.actionNumericAttributes));
          }
          if (banditLogData.actionNumericAttributes == null) {
            statement.setNull(11, Types.NULL);
          } else {
            // 👇 USED HERE
            statement.setString(11, EppoAttributes.serializeNonNullAttributesToJSONString(banditLogData.actionCategoricalAttributes));
          }

          statement.executeUpdate();
          System.out.println("Logged bandit assignment to bandit_test_bandit_assignments");
        } catch (SQLException e) {
          throw new RuntimeException("Unable to log bandit assignment "+e.getMessage(), e);
        }
      })
      .build();
    return EppoClient.init(config);
  }
```

As part of this, knowing if an `EppoValue` is a boolean is useful for serializing, so we've added back in the `EppoValue` `isXXX()` methods.

### CODE REVIEW

File: src/main/java/com/eppo/sdk/dto/EppoAttributes.java [hunk 1]
```diff
@@ -15,4 +19,62 @@ public EppoAttributes() {
...
+    return EppoAttributes.serializeAttributesToJSONString(this);
+  }
+
+  public static String serializeAttributesToJSONString(Map<String, ?> attributes) {
+    return EppoAttributes.serializeAttributesToJSONString(attributes, false);
+  }
+
+  public static String serializeNonNullAttributesToJSONString(Map<String, ?> attributes) {
+    return EppoAttributes.serializeAttributesToJSONString(attributes, true);
+  }
```
Comment (aarsilv) [inline src/main/java/com/eppo/sdk/dto/EppoAttributes.java:33]: Unsure yet if users will want to include or omit attributes with `null` values, so providing both options. (Our proxy omits them)

File: src/test/java/com/eppo/sdk/dto/EppoAttributesTest.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+          "boolean", EppoValue.valueOf(false),
+          "number", EppoValue.valueOf(1.234),
+          "string", EppoValue.valueOf("hello"),
+          "null", EppoValue.nullValue()
+        ));
+
+        String serializedJSONString = eppoAttributes.serializeToJSONString();
+        String expectedJson = "{ \"boolean\": false, \"number\": 1.234, \"string\": \"hello\", \"null\": null }";
+
+        JSONAssert.assertEquals(expectedJson, serializedJSONString, true);
```
Comment (aarsilv) [inline src/test/java/com/eppo/sdk/dto/EppoAttributesTest.java:24]: `JSONAssert()` ([link to library](https://github.com/skyscreamer/JSONassert)) is pretty slick. Tests for JSON equality so you don't have to worry about things like order.

File: pom.xml [hunk 3]
```diff
@@ -84,6 +84,12 @@
             <version>1.9.5</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.skyscreamer</groupId>
+            <artifactId>jsonassert</artifactId>
+            <version>1.5.1</version>
+            <scope>test</scope>
+        </dependency>
```
Comment (aarsilv) [inline pom.xml:92]: Test library for asserting JSON string equality ([link](https://github.com/skyscreamer/JSONassert))

File: src/test/java/com/eppo/sdk/dto/EppoAttributesTest.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    }
+
+    @Test
+    void testSerializeNumericAttributesToJSONString()  throws JSONException {
+        Map<String, Double> numericAttributes = new HashMap<>();
+        numericAttributes.put("positive", 12.3);
+        numericAttributes.put("negative", -45.6);
+        numericAttributes.put("integer", 43.0);
```
Comment (giorgiomartini0) [inline src/test/java/com/eppo/sdk/dto/EppoAttributesTest.java:38]: Maybe this was intentional and I'm missing something. I would have expected `43` here, not `43.0`, to check that Java ints are serialized correctly.

**Review by giorgiomartini0 — APPROVED** [review]
One tiny doubt; everything else looks great

File: src/test/java/com/eppo/sdk/dto/EppoAttributesTest.java (hunk 4 above)
Comment (aarsilv) [reply src/test/java/com/eppo/sdk/dto/EppoAttributesTest.java:38]: The way `Maps work, I _have_ to provide it a double 😞

Comment (giorgiomartini0) [reply src/test/java/com/eppo/sdk/dto/EppoAttributesTest.java:38]: Ah, true. Never mind!

**Pull request merged**